src/frontend/src/routes/+layout.svelte

```diff
@@ -82,6 +83,14 @@
 		const spinner = document.querySelector('body > #app-spinner');
 		spinner?.remove();
 	})();
+
+	/**
+	 * Event handler
+	 */
+
+	window.onpopstate = function () {
```

Comment:
- I'm confused about the goal of the implementation: Is the goal of the feature to call the `close` function of the modal on navigation back, or is it to treat the modal as an element of the navigation history? The implementation seems to address the former.

- Why not using `svelte:window`, which handles destroying the listener automatically?

- Nitpick: We use arrow functions.

- Not sure this belongs in the root layout. Maybe we should create a component like `Toasts` or `Busy` to scope this implementation? It would be easier to test that way.

- Maybe not for this PR, but if the application is `$isBusy`, should we prevent the modal from being closed?

Reply:
> yes, the goal is to close open modals when clicking on the browser back button to prevent undefined behaviour.

So just closing, right? For example, if a user browses to "Activity," then "Assets," and then opens the "Receive modal", what should happen when the user clicks back?

1. Close the modal and redirect the user to "Activity".
2. Close the modal and remain on the "Assets" page.

> I don't understand why

Because the main layout initalize the authentication, the analytics, the transations, synchronize the authentication, track some analytics, mount some workers for the authentication etc. etc. so sure it can be tested be you'll have to deal with all those things when you'll write a test.

Reply:
@peterpeterparker 

> So just closing, right? For example, if a user browses to "Activity," then "Assets," and then opens the "Receive modal", what should happen when the user clicks back?

It should do the first option. This means, closing the modal and redirect the user to the "Activity" page.

> Because the main layout initalize the authentication, the analytics...

Ok, i see. I will scope this implementation into its own component.

Reply:
@peterpeterparker I just introduced the `ModalExitHandler`.